market-generator/src/test/java/com/market/banica/generator/configuration/MarketConfigurationImplTest.java

```diff
@@ -0,0 +1,116 @@
+package com.market.banica.generator.configuration;
+
+import com.market.banica.generator.exception.NotFoundException;
+import com.market.banica.generator.model.GoodSpecification;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.api.io.TempDir;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.test.util.ReflectionTestUtils;
+
+import java.io.File;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+@ExtendWith(MockitoExtension.class)
+class MarketConfigurationImplTest {
+    private static final String ORIGIN_GOOD_PATTERN = "%s.%s";
+    private static final String TEST_ORIGIN = "europe";
+    private static final String TEST_GOOD = "bread";
+    private final Map<String, GoodSpecification> goods = new HashMap<>();
+
+    @TempDir
+    File file;
+
+    private File testFile;
+
+
+    //    @InjectMocks
+    private MarketConfigurationImpl marketConfiguration;
+
+
+    @BeforeEach
+    public void initialize() {
+        this.goods.put(String.format(ORIGIN_GOOD_PATTERN, TEST_ORIGIN, TEST_GOOD),
+                new GoodSpecification(TEST_GOOD,
+                        -1, 2, 3,
+                        4, 5, 6,
+                        9, 8, 9));
+        this.marketConfiguration = new MarketConfigurationImpl("src/test/java/com/market/banica/generator/property/market.properties");
+        ReflectionTestUtils.setField(marketConfiguration, "goods", goods);
+        this.testFile = new File(file, "testTest.properties");
+        ReflectionTestUtils.setField(marketConfiguration, "file", testFile);
+    }
+
+    @Test
+    public void addGoodSpecification_Should_Work() {
+        marketConfiguration.addGoodSpecification(TEST_ORIGIN, "meat",
+                -1, 2, 3,
+                4, 5, 6,
+                9, 8, 9);
+
+        assertEquals(2, goods.size());
+        assertTrue(this.testFile.length() > 0);
+        assertTrue(this.goods.containsKey(String.format(ORIGIN_GOOD_PATTERN, TEST_ORIGIN, TEST_GOOD)));
+        assertTrue(this.goods.containsKey(String.format(ORIGIN_GOOD_PATTERN, TEST_ORIGIN, "meat")));
+
+    }
+
+    @Test
```

Comment:
You can use @Test(expected = IllegalArgumentException.class) instead of assertThrows.